tartufo/scanner.py

```diff
@@ -476,3 +477,52 @@ def chunks(self):
         )
         for blob, file_path in self._iter_diff_index(diff_index):
             yield types.Chunk(blob, file_path)
+
+
+class FolderScanner(ScannerBase):
+    """Used to scan a folder."""
+
+    folder_options: types.FolderOptions
+    folder_path: str
+
+    def __init__(
+        self,
+        global_options: types.GlobalOptions,
+        folder_options: types.FolderOptions,
+        folder_path: str,
+    ) -> None:
+        """Used for scanning a folder.
+
+        :param global_options: The options provided to the top-level tartufo command
+        :param folder_options: The options specific to interacting with a folder
+        :param folder_path: The local filesystem path to scan
+        """
+        self.folder_options = folder_options
+        self.folder_path = folder_path
+        super().__init__(global_options)
+
+    @property
+    def chunks(self):
+        """Yield the individual file changes currently staged for commit.
+
+        :rtype: Generator[Chunk, None, None]
+        """
+
+        for blob, file_path in self._iter_folder():
+            if self.should_scan(file_path):
+                yield types.Chunk(blob, file_path)
+
+    def _iter_folder(self):
+        folder_path = pathlib.Path(self.folder_path)
+        for file_path in folder_path.glob(self.folder_options.pattern):
+            if file_path.is_file():
+                with open(file_path, "rb") as fhd:
+                    data = fhd.read()
+
+                try:
+                    blob = data.decode("utf-8")
+                except UnicodeDecodeError:
+                    # binary file, continue
+                    continue
```

Comment:
I'm wondering if this is really a 100% accurate test for a binary file. Does this work with files in other non-UTF8 characters sets such as CP1252? I think these are probably less common nowadays, but not non-existent. I also wonder about the performance of attempting to decode an entire file, vs just reading a few bytes from the top of the file to determine its type. How would this perform with, say, a several hundred megabyte file?

Reply:
Should I create a bug for this as well? https://github.com/godaddy/tartufo/blob/master/tartufo/scanner.py#L342

Reply:
IMHO this is bad practice checking checking the contents of an error message `if printable_diff.startswith("Binary files"):` as the error message should not be contractual.